src/main/java/com/contrastsecurity/maven/plugin/InstallAgentContrastMavenMojo.java

```diff
@@ -8,13 +8,32 @@
 
 import java.io.File;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 
 
 @Mojo(name = "install", defaultPhase = LifecyclePhase.VALIDATE, requiresOnline = true)
 public class InstallAgentContrastMavenMojo extends AbstractContrastMavenPluginMojo {
     String applicationName;
 
+    static Map<String, String> environmentToSessionMetadata = new TreeMap<String, String>();
+    static {
+        //Jenkins git plugin environment variables
```

Comment:
Sounds like you're creating this hash here with valid keys and dummy values. Then in `computeSessionMetadata` you're setting the values if they appear in the Env.

I'd rather 1) use null dummy values or 2) Make this an array of values and we assemble the hash (or never even assemble the hash at all) in `computeSessionMetadata`

Reply:
The values aren't dummy, they're the build property/'configuration key' types the environment variable should map to. (https://docs.contrastsecurity.com/user-vulnerableapps.html#session)

The first set of keys are those set by the Jenkins git plugin, and their respective values are the build property they should map to for the session metadata string.
The second set of keys are build number environment variables used by various CI systems that should map to the `buildNumber` session metadata key.

Reply:
For "dummy values" I meant this:

`environmentToSessionMetadata.put("GIT_COMMITTER_NAME", "committer");`

"committer" looks like a dummy value.

Reply:
Right, I know. `committer` In that instance isn't a dummy value though, it's the key under which the value of the environment variable `GIT_COMMITTER_NAME` should be put in the resulting metadata for the agents to consume.

Suppose the plugin is invoked under this environment
```
export GIT_COMMITTER_NAME='josh';
```

The metadata string would get built to be:
`-Dcontrast.application.session_metadata="committer=josh"`

Keys in this dict are environment variables we want to check for, and if they're found, we emit them as value (from the dict) = $variable

Reply:
Oh I see. That means the static initialize make a little more sense.